client/wp-admin-scripts/onboarding-tax-notice/index.js

```diff
@@ -28,30 +27,36 @@ const saveCompleted = () => {
 const showTaxCompletionNotice = () => {
 	const saveButton = document.querySelector( '.woocommerce-save-button' );
 
-	if ( saveButton.classList.contains( 'is-clicked' ) ) {
+	if ( saveButton.classList.contains( 'has-tax' ) ) {
 		return;
 	}
 
-	saveButton.classList.add( 'is-clicked' );
-	saveCompleted().then( () =>
-		dispatch( 'core/notices' ).createSuccessNotice(
-			__( "You've added your first tax rate!", 'woocommerce-admin' ),
-			{
-				id: 'WOOCOMMERCE_ONBOARDING_TAX_NOTICE',
-				actions: [
-					{
-						url: getAdminLink( 'admin.php?page=wc-admin' ),
-						label: __( 'Continue setup.', 'woocommerce-admin' ),
-					},
-				],
+	saveCompleted().then( () => {
+		// Check if a row was added successfully after WooCommerce removes invalid rows.
+		setTimeout( () => {
+			if ( ! document.querySelector( '.tips' ) ) {
+				return;
 			}
-		)
-	);
+			saveButton.classList.add( 'has-tax' );
+			dispatch( 'core/notices' ).createSuccessNotice(
+				__( "You've added your first tax rate!", 'woocommerce-admin' ),
+				{
+					id: 'WOOCOMMERCE_ONBOARDING_TAX_NOTICE',
+					actions: [
+						{
+							url: getAdminLink( 'admin.php?page=wc-admin' ),
+							label: __( 'Continue setup.', 'woocommerce-admin' ),
+						},
+					],
+				}
+			);
+		}, 500 );
+	} );
 };
 
-domReady( () => {
+window.onload = () => {
```

Comment:
Is there a particular reason you decided to use `window.onload` vs the `domReady` library?

Reply:
If we use `domReady`, `.tips` is still loading so the notice won't work. So I used `window.onload` to wait until `.tips` is loaded.

Reply:
Gotcha, I believe the biggest difference between `domReady` and `onload` is that it also waits for any assets to load (like images and such), which I guess allows enough of a delay.

That being said I think we should still stick with the original `domReady` use. Mostly because as I was testing I noticed that this script isn't loaded when a tax rate was added ([see here](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/OnboardingTasks.php#L261)), so we don't need it for that.
And the jquery script makes use of a global `htmlSettingsTaxLocalizeScript`, which contains a array of `rates`, so we could double check by using this.
`htmlSettingsTaxLocalizeScript && htmlSettingsTaxLocalizeScript.rates && htmlSettingsTaxLocalizeScript.rates.length > 0`